notebooks/04-esmvaltool-singularity/04-esmvaltool-singularity.ipynb

```diff
@@ -0,0 +1,285 @@
+    "\n",
+    "After successfully setting up the Jupyter server, please copy this notebook to the Spider file system. Then open it from the browser on your local PC (as specified in the above instructon). Effectively, this notebook run as a Slurm job on Spider."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bf44d734",
+   "metadata": {},
+   "source": [
+    "## Step 1: build a Singularity container\n",
+    "\n",
+    "ESMValTool is provided in the form of Docker containers. However on Spiders, [it is suggested to use Singularity](https://spiderdocs.readthedocs.io/en/latest/Pages/software_on_spider.html?highlight=singularity#singularity-containers) as the container technology. Therefore we will convert the existing Docker container to a Singularity container. Please run this command on Spiders:\n",
+    "\n",
+    "```sh\n",
+    "# Step1: build sif image (this should be done once)\n",
+    "singularity build esmvaltool_stable.sif docker://esmvalgroup/esmvaltool:stable\n",
+    "```\n",
+    "\n",
+    "This will build a Singularity Image File (.sif) named `esmvaltool_stable.sif` on the Spiders file system, from DockerHub. Note that this may take ~20 minutes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fbcc24a3",
+   "metadata": {},
+   "source": [
+    "## Step 2: User configeration\n",
+    "One can run the following command to apply user configurations for the ESMVALTool."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "43b7e268-b33f-4a54-ba1f-fec9fd8517f5",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2022-09-27 10:30:33,813 UTC [3531224] INFO    Creating folder /home/caroline-oku/.esmvaltool\n",
+      "2022-09-27 10:30:33,815 UTC [3531224] INFO    Copying file /opt/conda/envs/esmvaltool/lib/python3.10/site-packages/esmvalcore/config-user.yml to path /home/caroline-oku/.esmvaltool/config-user.yml.\n",
+      "2022-09-27 10:30:33,817 UTC [3531224] INFO    Copy finished.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Get user config file\n",
+    "!singularity run esmvaltool_stable.sif config get_config_user"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "48a0d67b",
+   "metadata": {},
+   "source": [
+    "## Step 3: Execute recipe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8443bb7e-aa48-4b81-98d3-9e8cc8df23f3",
+   "metadata": {},
+   "source": [
+    "We will execute two recipes under the folder `recipes`. Both of them plots a map of global temperature in January 2000, and plot a time series of mean annual temperature from 1850 to 2000. The difference of the two recepies are the climate datasets they use. In practice, if the dataset is huge, sequetially executing the two recipies will not be efficient.\n",
```

Comment:
Maybe add: "Analogous situations would be the execution of a computationally expensive recipe over a long time period , which could be split, and/or over large spatial extent at high resolution."